integrations/jenkins/main.py

```diff
@@ -57,6 +63,32 @@ async def on_resync_users(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
         yield users
 
 
+@ocean.on_resync(ObjectKind.STAGE)
+async def on_resync_stages(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    jenkins_client = init_client()
+    stages_count = 0
+    max_stages = 10000
+
+    stages_selector = cast(JenkinStagesResourceConfig, event.resource_config)
+    job_url = stages_selector.selector.job_url
+
+    logger.info(f"Syncing stages for job {job_url}")
+
+    async for stages in jenkins_client.get_stages(job_url):
+        logger.info(f"Received batch with {len(stages)} stages")
+        if stages_count + len(stages) > max_stages:
+            stages = stages[: max_stages - stages_count]
+            yield stages
+            logger.warning(
+                f"Reached the maximum limit of {max_stages} stages. Stopping the sync."
```

Comment:
your'e not really stopping the sync you just skipping the rest of the stages, so make this log more accurate

Reply:
noted